configuration/yaml.md

```diff
@@ -0,0 +1,644 @@
+
+> Tip: File naming: Use `.inc.yaml` or `.inc.yml` to prevent package files from being processed as standalone configurations.
+
+#### Package Example
+
+`main.yaml`
+
+```yaml
+version: 1
+
+packages:
+  livingroom-light: !include
+    file: templates/mqtt-light.inc.yaml
+    vars:
+      thingid: livingroom-light
+      name: Living_Room_Light
+      label: Living Room Light
+
+  # Reuse the same template
+  bedroom-light: !include
+    file: templates/mqtt-light.inc.yaml
+    vars:
+      thingid: bedroom-light
+      name: Bed_Room_Light
+      label: Bed Room Light
+
+# Custom override
+items:
+  Bed_Room_Light_Power:
+    label: My Bedroom Light Power
+    tags: [Switch2]
+    channels:
+      mqtt:topic:bedroom-light:switch2: {}
+```
+
+`templates/mqtt-light.inc.yaml`
+
+```yaml
+things:
+  mqtt:topic:${thingid}:
+    bridge: mqtt:broker:main
+    label: ${label}
+    channels:
+      switch1:
+        type: switch
+        config:
+          stateTopic: ${thingid}/switch/switch1/state
+          commandTopic: ${thingid}/switch/switch1/command
+
+items:
+  ${name}_Power:
+    type: Switch
+    label: ${label} Power
+    tags: [Switch]
+    # This can be done using the short form `channel:`
+    # but we're using the `channels:` map here to demonstrate merging
+    channels:
+      mqtt:topic:${thingid}:switch1: {}
+```
+
+➡️ Resolves to:
+
+```yaml
+things:
+  mqtt:topic:livingroom-light:
+    bridge: mqtt:broker:main
+    label: Living Room Light
+    channels:
+      switch1:
+        type: switch
+        config:
+          stateTopic: livingroom-light/switch/switch1/state
+          commandTopic: livingroom-light/switch/switch1/command
+
+  mqtt:topic:bedroom-light:
+    bridge: mqtt:broker:main
+    label: Bed Room Light
+    channels:
+      switch1:
+        type: switch
+        config:
+          stateTopic: bedroom-light/switch/switch1/state
+          commandTopic: bedroom-light/switch/switch1/command
+
+items:
+  Living_Room_Light_Power:
+    type: Switch
+    label: Living Room Light Power
+    tags: [Switch]
+    channel: mqtt:topic:livingroom-light:switch1
+
+  Bed_Room_Light_Power:
+    type: Switch
+    label: My Bedroom Light Power # Note this was overridden in the main file
+    tags: [Switch, Switch2] # merged from the main definition because it's a list
+    channels:
+      mqtt:topic:bedroom-light:switch1: {}
+      mqtt:topic:bedroom-light:switch2: {} # merged from the main definition because it's a map
+```
+
+🔄 Merge Behavior
+
+- Package contents are merged into the main YAML structure.
+- Scalar definitions in the main file override matching definitions from the package (by key).
+  This can be used to refine a generic package on a case-by-case basis.
+- Map and List definitions in the main file will be merged.
+
+File structure can be nested—e.g., you may organize packages under includes/ or templates/ as long as they’re excluded from default parsing by giving them a `.inc.yaml` or `.inc.yml` extension.
```

Comment:
Spaceless hyphen again.

Reply:
I think "includes/" and "templates/" should be in code quotes.

Reply:
> Spaceless hyphen again.

See above. They are rendered as emdash in the html - a slightly longer dash. I agree, here it's not strictly necessary. I'll change this to a comma.





> I think "includes/" and "templates/" should be in code quotes.

Done.